### PR TITLE
DEPR/BUG: fix Series.argmin/max

### LINKED ISSUE #16830: API: .argmax should be positional, not an alias for idxmax
In https://github.com/pandas-dev/pandas/issues/6214 it was reported that `argmax` changed (I think since Series didn't subclass `ndarray` anymore). It's sometimes useful to get the index position of the max, so it'd be nice if `.argmax` did that, leaving `idxmax` to always be label-based.

```python

In [1]: import pandas as pd
pd.Ser
In [2]: s = pd.Series([1, 2, 3, 0], index=['a', 'b', 'c', 'd'])

In [3]: s.idxmax()
Out[3]: 'c'

In [4]: s.argmax()
Out[4]: 'c'
```

I would expect `Out[4]` to be `2`. The current workaround is `np.argmax(x)`, which is OK, but doesn't generalize to DataFrames well.

We could deprecate `.argmax` in 0.21, pointing users to `.idxmax`.  Then later we can change `.argmax` to be always be positional.

---

Update: We've deprecated `.argmax/min`. Need to change the behavior next.

### PR DESCRIPTION
**Pull request by lphk92**
- [x] closes #16830
 - [x] tests added / passed
 - [x] passes ``git diff upstream/master -u -- "*.py" | flake8 --diff``
 - [ ] whatsnew entry

This PR actually implements the proper behavior for argmax() and argmin() for both Series and DataFrame. We will have to wait to rebase this onto master until after a new release.

### CODE REVIEW

File: pandas/tests/frame/test_analytics.py [hunk 1]
```diff
@@ -1031,6 +1031,34 @@ def test_idxmax(self):
...
+            for axis in [0, 1]:
+                for df in [frame, self.intframe]:
+                    result = df.argmin(axis=axis, skipna=skipna)
+                    expected = df.apply(Series.argmin, axis=axis,
+                                        skipna=skipna)
+                    tm.assert_series_equal(result, expected)
+
+        pytest.raises(ValueError, frame.argmin, axis=2)
+
+    def test_argmax(self):
```
Comment (jreback) [inline pandas/tests/frame/test_analytics.py]: use parametrize on these

File: doc/source/whatsnew/v0.21.0.txt [hunk 2]
```diff
@@ -116,6 +116,8 @@ Other API Changes
 Deprecations
 ~~~~~~~~~~~~
 - :func:`read_excel()` has deprecated ``sheetname`` in favor of ``sheet_name`` for consistency with ``.to_excel()`` (:issue:`10559`).
+- :method:`Series.argmax` has been deprecated in favor of :method:`Series.idxmax` (:issue:`16830`)
+- :method:`Series.argmin` has been deprecated in favor of :method:`Series.idxmin` (:issue:`16830`)
```
Comment (gfyoung) [inline doc/source/whatsnew/v0.21.0.txt]: Remove these.  Refrain from writing anything in the `whatsnew` until we can agree which version your changes will be applied.

**Comment by gfyoung** [comment]
Tagging this for `1.0` unless we are doing another 0.2x version after 0.21

File: pandas/tests/frame/test_analytics.py (hunk 1 above)
Comment (lphk92) [inline pandas/tests/frame/test_analytics.py]: These new functions were written to follow the existing structure of the idxmin and idxmax functions. Would you like me to parametrize those functions as well so as to not introduce an inconsistency in the testing methodology?

Comment (gfyoung) [inline pandas/tests/frame/test_analytics.py]: No, he's referring to the fact that you were using nested for-loops for testing.  You can avoid those by parametrizing as illustrated <a href="https://docs.pytest.org/en/latest/parametrize.html">here</a>

**Comment by jreback** [comment]
I would rather open an issue that references this and mark it for 1.0. Then the PR can be resurrected at that time.

**Pull request closed without merging**